src/main/java/org/folio/search/model/metadata/PostProcessResourceDescriptionConverter.java

```diff
@@ -0,0 +1,71 @@
+package org.folio.search.model.metadata;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
+import com.fasterxml.jackson.databind.util.StdConverter;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * This Jackson converter is used to post-process ResourceDescription after it has been deserialized.
+ * It does following actions:
+ * * Resolves types for object/plain fields. Type must be defined in {@code fieldTypes} section of the
+ * resource description and than it can be referenced in actual type via $type: [type] property.
+ * * Builds flattened map of field path and field description pairs.
+ */
+public class PostProcessResourceDescriptionConverter extends StdConverter<ResourceDescription, ResourceDescription> {
+  @Override
+  public ResourceDescription convert(ResourceDescription value) {
+    resolveFieldByType(value, value.getFields());
+    value.setFlattenFields(flattenFields(value));
+
+    return value;
+  }
+
+  private void resolveFieldByType(ResourceDescription desc, Map<String, FieldDescription> fields) {
+    for (var entry : fields.entrySet()) {
+      if (isNotBlank(entry.getValue().getFieldType())) {
+        entry.setValue(getFieldByType(desc, entry.getValue().getFieldType()));
+      }
+
+      // Resolve nested properties recursively
+      if (entry.getValue() instanceof ObjectFieldDescription) {
+        resolveFieldByType(desc, ((ObjectFieldDescription) entry.getValue()).getProperties());
```

Comment:
`entry.getValue()` is used 4 times, I guess it is worth to extract it to variable

Reply:
Fixed